torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -236,6 +242,24 @@ def init_dtype_attrs(self, mp_policy: MixedPrecisionPolicy):
         self.reduce_dtype = reduce_dtype
         # None indicates that the mixed precision is not enabled
 
+    def _init_extensions(self) -> None:
+        inner_tensor = self._sharded_local_tensor
+        has_fsdp_pre_all_gather = hasattr(inner_tensor, "fsdp_pre_all_gather")
```

Comment:
Note: I am thinking to always check this via `hasattr(inner_tensor, "fsdp_pre_all_gather")` to leave a backdoor if somehow we can monkey patch this method onto the tensor without making it a subclass.